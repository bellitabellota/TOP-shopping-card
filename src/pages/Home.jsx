import styles from "./Home.module.css"
import { useOutletContext } from "react-router-dom";


function Home() {
  const [products, addProductToCart] = useOutletContext();

  const productCards = products.map((product) => {
    const onChangeHandler = (e) => {
      const inputField = document.querySelector("input:focus");
      inputField.value = e.target.value;
    }

    return <div className={styles.productCard} key={product.id} data-product-id={product.id}>
              <img src={product.image} alt={product.title} />
              <div className={styles.productText}>
                <div>
                  <h2>{product.title}</h2>
                  <p className={styles.productDescription}>{product.description}</p>
                </div>
                <div className={styles.cardFooter}>
                  <div className={styles.quantityContainer}>
                    <input type="number" defaultValue={1} onChange={onChangeHandler}/>
                    <button onClick={()=> {addProductToCart(product)}}>Add To Cart</button>
                  </div>
                  <p className={styles.productPrice}>{product.price} EUR</p>
                </div>
              </div>
            </div>
  })

  return (
    <>
      <h1>All Products</h1>
      <main>
       {productCards}
      </main>
    </>
  )
}

export default Home;