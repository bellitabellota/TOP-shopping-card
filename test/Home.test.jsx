import { render } from "@testing-library/react";
import Home from "../src/pages/Home"
import { describe, it, vi, expect } from "vitest";
import { useOutletContext } from "react-router-dom";

vi.mock("react-router-dom", () => ({
  useOutletContext: vi.fn(),
}));

describe("Home", () => {
  it("renders a list of products", () => {
    const productsMock = [{
      id: 1,
      title: "Test Product 1",
      price: 109.95,
      description: "The Test Product 1 description ...",
      category: "test products",
      image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
      rating: {
        rate: 3.9,
        count: 120
      }
    },
    {
      id: 2,
      title: "Test Product 2",
      price: 22.3,
      description: "The Test Product 2 description ...",
      category: "test products",
      image: "https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg",
      rating: {
        rate: 4.1,
        count: 259
      }
    }]

    const addProductToCartMock = vi.fn();

    useOutletContext.mockReturnValue([productsMock, addProductToCartMock]);

    const {container} = render(<Home />);

    expect(container).toMatchSnapshot();
  })
})