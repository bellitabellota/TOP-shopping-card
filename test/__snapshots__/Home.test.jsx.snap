// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > renders a list of products 1`] = `
<div>
  <h1>
    All Products
  </h1>
  <div
    class="_cardsContainer_4dfadf"
  >
    <div
      class="_productCard_4dfadf"
      data-product-id="1"
    >
      <img
        alt="Test Product 1"
        src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
      />
      <div
        class="_productText_4dfadf"
      >
        <div>
          <h2>
            Test Product 1
          </h2>
          <p
            class="_productDescription_4dfadf"
          >
            The Test Product 1 description ...
          </p>
        </div>
        <div
          class="_cardFooter_4dfadf"
        >
          <div
            class="_quantityContainer_4dfadf"
          >
            <input
              type="number"
              value="1"
            />
            <button>
              Add To Cart
            </button>
          </div>
          <p
            class="_productPrice_4dfadf"
          >
            109.95
             EUR
          </p>
        </div>
      </div>
    </div>
    <div
      class="_productCard_4dfadf"
      data-product-id="2"
    >
      <img
        alt="Test Product 2"
        src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
      />
      <div
        class="_productText_4dfadf"
      >
        <div>
          <h2>
            Test Product 2
          </h2>
          <p
            class="_productDescription_4dfadf"
          >
            The Test Product 2 description ...
          </p>
        </div>
        <div
          class="_cardFooter_4dfadf"
        >
          <div
            class="_quantityContainer_4dfadf"
          >
            <input
              type="number"
              value="1"
            />
            <button>
              Add To Cart
            </button>
          </div>
          <p
            class="_productPrice_4dfadf"
          >
            22.3
             EUR
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
